### PR TITLE
FEATURE: SETI-668 Support custom name of status in zuul pipelines

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

File: doc/source/zuul.rst [hunk 1]
```diff
@@ -591,6 +591,10 @@ explanation of each of the parameters::
   unsuccessful change merges.
   Default: ``2``.
 
+**status-name**
+  Describes what status name should Zuul report back to Github
```
Comment (michalvanco) [inline doc/source/zuul.rst]: this doc is quite generat, I don't know if there is a "gerrit-like" equivalent of this

Comment (janpriessnitz) [inline doc/source/zuul.rst]: Zuul reports only a review message to Gerrit, so there isn't really an equivalent. I'll rename the attribute to github-status-name

**Comment by janpriessnitz** [comment]
Done. Conflicts with another PR, but should be easy to resolve.

File: tests/fixtures/layout.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ includes:
 pipelines:
   - name: check
     manager: IndependentPipelineManager
+    github-status-name: check
```
Comment (michalvanco) [inline tests/fixtures/layout.yaml]: remove? to properly test the "default" value?

File: tests/test_github.py [hunk 3]
```diff
@@ -499,3 +499,25 @@ def test_failed_change_at_head(self):
...
+
+        self.assertIn('check', A.statuses)
+
+    def test_custom_github_status_name(self):
+        "Tests that Zuul reports status with custom name when defined"
+        self.config.set('zuul', 'layout_config',
+                        'tests/fixtures/layout-github-status-name.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+        github_status_name = 'abcd'
```
Comment (michalvanco) [inline tests/test_github.py]: just a cosmetic...I would remove this line and put the `abcd` directly in https://github.com/gooddata/zuul/pull/32/files#diff-f8a6c81b3074b45c308c6126516948bdR523

**Comment by michalvanco** [comment]
```
Ran 253 tests in 207.949s
PASSED (id=0)
Slowest Tests
Test id                                                                Runtime (s)
---------------------------------------------------------------------  -----------
tests.test_cloner.TestCloner.test_upgrade                              16.312
tests.test_scheduler.TestScheduler.test_idle                           12.622
tests.test_scheduler.TestScheduler.test_crd_check_transitive           11.469
tests.test_scheduler.TestScheduler.test_failing_dependent_changes       9.546
tests.test_scheduler.TestScheduler.test_merge_failure_reports           8.619
tests.test_scheduler.TestScheduler.test_dependent_behind_dequeue        8.506
tests.test_scheduler.TestScheduler.test_merger_repack_large_change      8.420
tests.test_requirements.TestRequirements.test_pipeline_require_reject   8.388
tests.test_scheduler.TestScheduler.test_dependent_changes_dequeue       8.386
tests.test_scheduler.TestScheduler.test_job_from_templates_launched     7.565
```
Thank you @janpriessnitz!!!

**Pull request merged**